clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -56,6 +58,7 @@
 public class SubscriptionState {
     private static final String SUBSCRIPTION_EXCEPTION_MESSAGE =
             "Subscription to topics, partitions and pattern are mutually exclusive";
+    private static final Logger log = LoggerFactory.getLogger(SubscriptionState.class);
```

Comment:
Can we pass `LogContext` from the consumer to construct this? Often applications have many consumers and it becomes hard to distinguish individual consumer behavior, so we created a log context which basically just adds a prefix to all log messages.

Reply:
Done.